### PR TITLE
Building Forms with Ruby on Rails

### PR DESCRIPTION
**Pull request by emasdev**
In this project, we built a form the old fashioned way and then the Rails way

- Generated the model/controller/migrations for Users.
- Validate input data from form new/edit User form.

### CODE REVIEW

**Review by mikenath223 — CHANGES_REQUESTED** [review]
### Hi Team 👋

You've both done well on this.

However, kindly check the commented code below for changes required. 👇

In addition, kindly fix these issues and request a new code review.

**HAPPY CODING ✨**

File: app/controllers/users_controller.rb [hunk 1]
```diff
@@ -0,0 +1,36 @@
+class UsersController < ApplicationController
+  def new
+    @user = User.new
+  end
+
+  def create
+    # @user = User.new(username: params[:username], email: params[:email], password: params[:password])
```
Comment (mikenath223) [inline app/controllers/users_controller.rb]: - [x] Please remove the commented code.

File: app/views/users/new.html.erb [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+  <%= "Email #{@user.errors.messages[:email].first}" if !@user.errors.messages[:email].empty? %><br><br>
+  
+  <%= f.label "Password" %>
+  <%= f.password_field :password %>
+  <%= "Password #{@user.errors.messages[:password].first}" if !@user.errors.messages[:password].empty? %><br><br>
+  <%= f.submit "Create" %>
+<% end %>
+
+<!--
+<%= form_tag("/users", method: "post") do %>
```
Comment (mikenath223) [inline app/views/users/new.html.erb]: - [x] Kindly use the users_path helper here instead.

File: app/views/users/new.html.erb [hunk 3]
```diff
@@ -0,0 +1,52 @@
...
+    <%= form.label "Email" %>
+    <%= form.text_field :email %><br><br>
+    <%= form.label "Password" %>
+    <%= form.text_field :password %><br><br>
+    <%= form.submit "Create" %>
+  <% end %>
+-->
+
+<!--
+<form action="/users" method="post" accept-charset="UTF-8">
```
Comment (mikenath223) [inline app/views/users/new.html.erb]: - [x] Use the users_path helper here too.

File: app/views/users/new.html.erb [hunk 4]
```diff
@@ -0,0 +1,52 @@
...
+<% end %>
+
+<!--
+<%= form_tag("/users", method: "post") do %>
+  <%= label_tag(:username, "Name:") %>
+  <%= text_field_tag(:username) %> <br><br>
+    <%= label_tag(:email, "Email:") %>
+  <%= email_field_tag(:email) %> <br><br>
+    <%= label_tag(:password, "Password:") %>
+  <%= password_field_tag(:password) %> <br><br>
```
Comment (mikenath223) [inline app/views/users/new.html.erb]: - [x] For each of the label tag here please change it to this format:  `<%= email_field_tag 'email' %>`

File: app/views/users/new.html.erb [hunk 5]
```diff
@@ -0,0 +1,52 @@
+<h2>Create User</h2>
+
+<%= form_for @user, url: {action: "create"} do |f| %>
```
Comment (mikenath223) [inline app/views/users/new.html.erb]: - [x]  No need to add a url here the @user variable will be enough.

**Comment by MarilenaRoque** [comment]
## Hello @mikenath223. Thanks for the review! 

We're happy to say that all the requirements were applied:

- [x] Used _path helper 

 - [x] Removed the commented code.

- [x] For each of the label tag we changed it to this format: <%= email_field_tag 'email' %>

- [x] Removed url from form_for tag.

**Comment by Dandush03** [comment]
# Status: :rocket: Project Approved :rocket: 
## Great job applying does changes
# Happy Coding! :computer: :nerd_face:

**Pull request merged**